vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1,1017 +1,1023 @@
-        parentRequestDebouncer.cancel();
-      }
-      ensureSubCacheDebouncer = undefined;
-      parentRequestDebouncer = undefined;
-      ensureSubCacheQueue = [];
-      parentRequestQueue = [];
-      updateAllGridRowsInDomBasedOnCache();
-    };
-
-    const deleteObjectContents = function(obj) {
-      let props = Object.keys(obj);
-      for (let i = 0; i < props.length; i++) {
-        delete obj[props[i]];
-      }
-    }
-
-    grid.$connector.updateSize = function(newSize) {
-      grid.size = newSize;
-    };
-
-    grid.$connector.updateUniqueItemIdPath = function(path) {
-      grid.itemIdPath = path;
-    }
-
-    grid.$connector.expandItems = function(items) {
-      let newExpandedItems = Array.from(grid.expandedItems);
-      items.filter(item => !grid._isExpanded(item))
-        .forEach(item =>
-          newExpandedItems.push(item));
-      grid.expandedItems = newExpandedItems;
-    }
-
-    grid.$connector.collapseItems = function(items) {
-      let newExpandedItems = Array.from(grid.expandedItems);
-      items.forEach(item => {
-        let index = grid._getItemIndexInArray(item, newExpandedItems);
-        if(index >= 0) {
-            newExpandedItems.splice(index, 1);
+      const isSelectedOnGrid = tryCatchWrapper(function(item) {
+        const selectedItems = grid.selectedItems;
+        for(let i = 0; i < selectedItems; i++) {
+          let selectedItem = selectedItems[i];
+          if (selectedItem.key === item.key) {
+            return true;
+          }
         }
+        return false;
+      })
+
+      grid.$connector.reset = tryCatchWrapper(function() {
+        grid.size = 0;
+        deleteObjectContents(cache);
+        deleteObjectContents(grid._cache.items);
+        deleteObjectContents(lastRequestedRanges);
+        if(ensureSubCacheDebouncer) {
+          ensureSubCacheDebouncer.cancel();
+        }
+        if(parentRequestDebouncer) {
+          parentRequestDebouncer.cancel();
+        }
+        ensureSubCacheDebouncer = undefined;
+        parentRequestDebouncer = undefined;
+        ensureSubCacheQueue = [];
+        parentRequestQueue = [];
+        updateAllGridRowsInDomBasedOnCache();
       });
-      grid.expandedItems = newExpandedItems;
-      items.forEach(item => grid.$connector.removeFromQueue(item));
-    }
-
-    grid.$connector.removeFromQueue = function(item) {
-      let itemId = grid.getItemId(item);
-      delete treePageCallbacks[itemId];
-      grid.$connector.removeFromArray(ensureSubCacheQueue, item => item.itemkey === itemId);
-      grid.$connector.removeFromArray(parentRequestQueue, item => item.parentKey === itemId);
-    }
-
-    grid.$connector.removeFromArray = function(array, removeTest) {
-      if(array.length) {
-        for(let index = array.length - 1; index--; ) {
-           if (removeTest(array[index])) {
-             array.splice(index, 1);
-           }
+
+      const deleteObjectContents = tryCatchWrapper(function(obj) {
```

Comment:
This seems to only be called from within other methods that are already using `tryCatchWrapper` so it could be removed here.

Reply:
Updated.

Reply:
Looks like nothing was updated here, it's still using `tryCatchWrapper`?

Reply:
`grid.$connector.reset` was using `tryCatchWrapper` before, but it seems to have been removed from there in the latest updates. This `deleteObjectContents` is called from `grid.$connector.reset`.

I think it's better to have those `$connector` methods use the wrapper and avoid the wrapper in these `const` methods if not otherwise necessary.

So I'd put the wrapper back on `grid.$connector.reset` and remove it from here (`deleteObjectContents`).

Reply:
I guess you looked at the wrong line (when removing it based on my comment) or you misunderstood which line I wanted to change.

Reply:
Updated to wrap the reset.